src/docfx/abstractions/IRedirectionProvider.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Docs.Build
+{
+    internal interface IRedirectionProvider
+    {
+        bool TryGetRedirection(string sourcePath, out string redirectUrl);
+
+        bool TryGetRename(string sourcePath, out string originalPath);
```

Comment:
From the usage, TryGetRename is to get document id, should we directly provide TryGetRedirectionDocumentId()?